app/models/solidus_bolt/gateway.rb

```diff
@@ -4,9 +4,29 @@ module SolidusBolt
   class Gateway
     def initialize(options = {}); end
 
-    def authorize(_amount, _payment_source, _gateway_options)
-      binding.pry
-      raise NotImplementedError, 'authorize method has not been implemented in SolidusBolt::Gateway class'
+    def authorize(amount, payment_source, gateway_options)
+      order_number = gateway_options[:order_id].split('-').first
+      order = Spree::Order.find_by(number: order_number)
+
+      authorization_response = ::SolidusBolt::Transactions::AuthorizeService.call(
+        order: order,
+        create_bolt_account: payment_source.create_bolt_account,
+        credit_card: credit_card_params(payment_source)
+      )
+
+      payment_source.update(
+        transaction_id: authorization_response['id'],
```

Comment:
I see you are saving a lot of `transaction_*` fields on the payment_source, are these unique for transactions? If yes, they shouldn't be stored on the payment_source, because the payment source reflects the source of the payment used by the user, not the payment.
They would have been stored to the Spree::Payment